kong.conf.default

```diff
@@ -1989,3 +1989,88 @@
+
+# The Nginx Wasm module has its own set of directives which can be tuned via
+# special `nginx_wasm_*` directives.
+#
+# The following namespaces are supported:
+#
+# - `nginx_wasm_<directive>`: Injects `<directive>` into the `wasm {}` block.
+# - `nginx_wasm_shm_<name>`: Injects `shm_kv <name>` into the `wasm {}` block,
+#    allowing operators to define custom shared memory zones which are usable by
+#    the `get_shared_data`/`set_shared_data` Proxy-Wasm SDK functions.
+# - `nginx_wasm_wasmtime_<flag>`: Injects `flag <flag>` into the `wasmtime {}`
+#    block, allowing various Wasmtime-specific flags to be set.
+#
+# The documentation for all supported directives can be found in the Nginx Wasm
+# module repository:
+#
+# https://github.com/Kong/ngx_wasm_module/blob/main/docs/DIRECTIVES.md
+#
+# Additionally, there are several Wasm-specific directives which can be tuned
+# specifically at the `http {}` or `server {}` level. These are supported via
+# the `nginx_http_<directive>` and `nginx_proxy_<directive>` namespaces:
+#
+# - `http {}`
+#   * `nginx_http_wasm_socket_connect_timeout`
+#   * `nginx_http_wasm_socket_read_timeout`
+#   * `nginx_http_wasm_socket_send_timeout`
+#   * `nginx_http_wasm_socket_buffer_size`
+# - `server {}`
+#   * `nginx_proxy_wasm_socket_connect_timeout`
+#   * `nginx_proxy_wasm_socket_read_timeout`
+#   * `nginx_proxy_wasm_socket_send_timeout`
+#   * `nginx_proxy_wasm_socket_buffer_size`
+#
+# The values for these settings are inherited from their `nginx_*_lua_*`
+# counterparts if they have not been explicitly set. For instance, if you set
+# `nginx_http_lua_socket_connect_timeout` in your `kong.conf` file, the value
+# of this setting will be propagated to `nginx_http_wasm_socket_connect_timeout`
+# unless you _also_ set `nginx_http_wasm_socket_connect_timeout`.
+#
+# Some TLS-related settings receive special treatment as well:
+#
+# - `lua_ssl_trusted_certificate`: when set, the value is  propagated to the
+#   `nginx_wasm_tls_trusted_certificate` directive.
+# - `lua_ssl_verify_depth`: when set (to a value greater than zero), several
+#   TLS-related `nginx_wasm_*` settings are enabled:
+#   * `nginx_wasm_tls_verify_cert`
+#   * `nginx_wasm_tls_verify_host`
+#   * `nginx_wasm_tls_no_verify_warn`
+#
+# Like other `kong.conf` fields, all injected Nginx directives documented here
+# can be set via environment variable. For instance, setting:
+#   `KONG_NGINX_WASM_TLS_VERIFY_CERT=<value>`
+#
+#   Will inject the following in to the `wasm {}` block:
+#
+#   `tls_verify_cert <value>;`
+#
+#
+# There are several Nginx directives supported by `ngx_wasm_module` which are
+# unsupported by or irrelevant to Kong. These directives should not be used and
+# may be rejected or silently dropped in future versions of Kong:
+#
+# - `wasm_call`
+# - `resolver`
+# - `resolver_add`
+# - `proxy_wasm_request_headers_in_access`
+
+#nginx_http_proxy_wasm_lua_resolver = on
```

Comment:
Should this be a documented flag if it's not meant to be tinkered with?...

Reply:
Unsure of this myself. I wanted some stern advice in here in case folks read the ngx_wasm_module docs and start to think that they can tinker with _everything_, so if it's not explicitly documented here I think I would have still put it on a "probably don't" list.